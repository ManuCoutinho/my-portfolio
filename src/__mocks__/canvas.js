HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
