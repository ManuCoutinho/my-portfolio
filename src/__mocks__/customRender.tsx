import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
