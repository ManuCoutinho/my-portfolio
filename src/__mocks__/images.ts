export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
