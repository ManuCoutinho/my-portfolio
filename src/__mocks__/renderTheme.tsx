import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
