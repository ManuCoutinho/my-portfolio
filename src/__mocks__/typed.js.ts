/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-empty-function */

export default class Typed {
  constructor(elementId: string, options: any) {}
  toggle(): void {}
  stop(): void {}
  start(): void {}
  destroy(): void {}
  reset(restart?: boolean): void {}
}
