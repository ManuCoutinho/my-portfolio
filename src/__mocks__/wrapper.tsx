/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
