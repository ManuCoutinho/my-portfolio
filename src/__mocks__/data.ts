import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
