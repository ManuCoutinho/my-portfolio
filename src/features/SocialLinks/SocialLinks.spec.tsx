import { screen } from '@testing-library/react'
import { render } from '__mocks__/customRender'
import { SocialLinks } from '.'

describe('<SocialLinks/>', () => {
	it('should render with all links', () => {
		render(<SocialLinks direction='row' />)

		expect(screen.getByRole('link', { name: /email/i })).toBeInTheDocument()
		expect(screen.getByRole('link', { name: /linkedin/i })).toBeInTheDocument()
		expect(screen.getByRole('link', { name: /github/i })).toBeInTheDocument()
		expect(screen.getByRole('link', { name: /dev.to/i })).toBeInTheDocument()
		expect(screen.getByRole('link', { name: /twitter/i })).toBeInTheDocument()
	})
	it('should render without email and twitter icons', () => {
		render(<SocialLinks direction='row' hideIcon={['twitter', 'email']} />)

		expect(
			screen.queryByRole('link', { name: /email/i })
		).not.toBeInTheDocument()
		expect(
			screen.queryByRole('link', { name: /twitter/i })
		).not.toBeInTheDocument()
		expect(screen.queryByRole('switch')).not.toBeInTheDocument()
		expect(screen.getByRole('link', { name: /github/ })).toBeInTheDocument()
	})
	it('should render with toggle icon', () => {
		render(<SocialLinks direction='column' showToggle />)
		expect(screen.getByRole('switch')).toBeInTheDocument()
	})
	it('should match to snapshot', () => {
		const { container } = render(<SocialLinks direction='row' />)

		expect(container).toMatchSnapshot()
	})
})
