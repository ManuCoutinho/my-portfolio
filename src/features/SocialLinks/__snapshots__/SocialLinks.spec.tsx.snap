// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLinks/> should match to snapshot 1`] = `
<div>
  <nav
    aria-label="social links"
    class="sc-3638c0c7-0 zEqDV"
    data-state="secondary"
    direction="row"
    role="navigation"
  >
    <a
      aria-label="Acessar linkedin"
      class="sc-3638c0c7-1 fteKrd"
      href="https://linkedin.com/in/emanuela-coutinho"
      rel="noopener noreferrer"
      target="_blank"
      title="linkedin"
    >
      <iconify-icon
        icon="ph:linkedin-logo"
      />
    </a>
    <a
      aria-label="Acessar github"
      class="sc-3638c0c7-1 fteKrd"
      href="https://github.com/manucoutinho"
      rel="noopener noreferrer"
      target="_blank"
      title="github"
    >
      <iconify-icon
        icon="iconoir:github-outline"
      />
    </a>
    <a
      aria-label="Acessar twitter"
      class="sc-3638c0c7-1 fteKrd"
      href="https://twitter.com/manucout"
      rel="noopener noreferrer"
      target="_blank"
      title="twitter"
    >
      <iconify-icon
        icon="radix-icons:twitter-logo"
      />
    </a>
    <a
      aria-label="Acessar email"
      class="sc-3638c0c7-1 fteKrd"
      href="mailto:devmanucoutinho@gmail.com"
      rel="noopener noreferrer"
      target="_blank"
      title="email"
    >
      <iconify-icon
        icon="ep:message"
      />
    </a>
    <a
      aria-label="Acessar dev.to"
      class="sc-3638c0c7-1 fteKrd"
      href="https://dev.to/manucoutinho"
      rel="noopener noreferrer"
      target="_blank"
      title="dev.to"
    >
      <iconify-icon
        icon="ph:dev-to-logo"
      />
    </a>
  </nav>
</div>
`;
